### PR TITLE
[MAINTENANCE] Rename GX Cloud "contract" resource to "checkpoint"

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Due to changes made in the GX Cloud backend, we can rename "contract" to the more apt "checkpoint" moving forward.

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/data_context/store/ge_cloud_store_backend.py [hunk 1]
```diff
@@ -34,7 +34,8 @@ class GeCloudRESTResource(str, Enum):
 
 class GeCloudStoreBackend(StoreBackend, metaclass=ABCMeta):
     PAYLOAD_ATTRIBUTES_KEYS: Dict[GeCloudRESTResource, str] = {
-        GeCloudRESTResource.CONTRACT: "checkpoint_config",
+        GeCloudRESTResource.CHECKPOINT: "checkpoint_config",
+        GeCloudRESTResource.CONTRACT: "checkpoint_config",  # Deprecated
```
Comment (cdkini) [inline great_expectations/data_context/store/ge_cloud_store_backend.py]: We need to keep this around as our E2E tests rely on old behavior - leaving a generic comment here but just wanted to flag for discussion before we move forward.

File: great_expectations/data_context/store/ge_cloud_store_backend.py [hunk 2]
```diff
@@ -57,7 +58,7 @@ class GeCloudStoreBackend(StoreBackend, metaclass=ABCMeta):
         **{
             GeCloudRESTResource.BATCH: "batches",
             GeCloudRESTResource.CHECKPOINT: "checkpoints",
-            GeCloudRESTResource.CONTRACT: "contracts",
+            GeCloudRESTResource.CONTRACT: "contracts",  # Deprecated
```
Comment (cdkini) [inline great_expectations/data_context/store/ge_cloud_store_backend.py]: We need to keep this around as our E2E tests rely on old behavior - leaving a generic comment here but just wanted to flag for discussion before we move forward.

**Review by Kilo59 — APPROVED** [review]

File: tests/data_context/cloud_data_context/test_checkpoint_crud.py [hunk 3]
```diff
@@ -118,53 +117,53 @@ def _mocked_get_response(*args, **kwargs):
                         "organization_id": f"{organization_id}",
                         "updated_at": "2022-08-02T17:55:45.107550",
                     },
-                    "id": contract_id,
+                    "id": checkpoint_id,
                     "links": {
-                        "self": f"/organizations/{organization_id}/contracts/{contract_id}"
+                        "self": f"/organizations/{organization_id}/checkpoints/{checkpoint_id}"
                     },
                     "relationships": {
```
Comment (wookasz) [inline tests/data_context/cloud_data_context/test_checkpoint_crud.py]: note: I don't think you're using these, but we're going to drop `relationships` in all our future APIs

File: tests/data_context/cloud_data_context/test_checkpoint_crud.py [hunk 4]
```diff
@@ -118,53 +117,53 @@ def _mocked_get_response(*args, **kwargs):
...
-                                "self": f"/organizations/{organization_id}/contracts/{contract_id}/relationships/suite-validation-results",
+                                "related": f"/organizations/{organization_id}/checkpoint/{checkpoint_id}/suite-validation-results",
+                                "self": f"/organizations/{organization_id}/checkpoints/{checkpoint_id}/relationships/suite-validation-results",
                             }
                         },
                     },
-                    "type": "contract",
+                    "type": "checkpoint",
                 },
                 "jsonapi": {"version": "1.0"},
```
Comment (wookasz) [inline tests/data_context/cloud_data_context/test_checkpoint_crud.py]: note: I don't think you're using these, but we're going to drop `jsonapi` in all our future APIs

File: tests/data_context/cloud_data_context/test_checkpoint_crud.py [hunk 5]
```diff
@@ -118,53 +117,53 @@ def _mocked_get_response(*args, **kwargs):
...
                 "links": {
```
Comment (wookasz) [inline tests/data_context/cloud_data_context/test_checkpoint_crud.py]: note: I don't think you're using these, but we're going to drop `links` in all our future APIs

**Review by wookasz — APPROVED** [review]

**Pull request merged**